docs/esterni/manuale_installazione/manuale_installazione.tex

```diff
@@ -0,0 +1,285 @@
+\begin{itemize}
+  \item la mnemonic phrase del proprio wallet;
+  \item il provider ottenuto al punto \hyperref[sec:provider]{2.1.2}.
+\end{itemize}
+Di seguito un esempio del formato corretto da utilizzare:
+\begin{verbatim}
+MNEMONIC=mnemonic phrase del proprio wallet
+PROVIDER=provider url
+\end{verbatim}
+\subsection{Eseguire lo script}
+Una volta soddisfatti i requisiti e ultimata la configurazione è possibile procedere con l'esecuzione dello script python. Verranno illustrati due modi per la corretta esecuzione dello script \textit{sell.py}.
+\subsubsection{Sell.py da terminale}
+È possibile eseguire lo script \textit{sell.py} direttamente da terminale, attraverso il comando:
+\begin{lstlisting}[language=bash]
+  $ python3 sell.py [item price]
+\end{lstlisting}
+Sostituire \verb|[item price]| con il prezzo con il quale si desidera vendere il proprio prodotto.
+Una volta eseguito lo script, verrà ritornato e stampato l'id e il prezzo del prodotto messo in vendita.
+\subsubsection{Script python}
+È possibile realizzare un proprio script in python per la vendita di prodotti.\\
+La funzione per mettere in vendita il proprio prodotto è \textbf{sell\_item} di \textit{sell.py}. La funzione richiede in input un solo parametro, ovvero il prezzo in dollari del prodotto che si desidera vendere.\\
+È dunque necessario includere nel proprio script la funzione come segue:
+\begin{verbatim}
+from sell import sell_item
+\end{verbatim}
+\subsection{Test}
+Per eseguire il test sullo script è sufficiente eseguire il seguente comando:
+\begin{lstlisting}[language=bash]
+$ pytest
+\end{lstlisting}
+\section{Contract}
+\subsection{Requisiti}
+\subsubsection{Truffle, plugin e librerie}
+Per compilare lo smart contract e deployarlo è necessario installare Truffle e alcuni plugin e librerie all'interno della cartella "Contract". Eseguire i seguenti comandi per una corretta installazione:
+\begin{lstlisting}[language=bash]
+  $ npm install
+  $ npm install -g truffle
+  $ npm install @truffle/hdwallet-provider
+\end{lstlisting}
+per le librerie eseguire:
+\begin{lstlisting}[language=bash]
+  $ npm install @openzeppelin/contracts
+  $ npm install @chainlink/contracts
+\end{lstlisting}
+infine per la verifica e test installare:
+\begin{lstlisting}[language=bash]
+  $ npm install truffle-plugin-verify
+  $ npm install solidity-coverage
+\end{lstlisting}
+\subsection{Configurazione}
+È necessaria la creazione di tre file nella cartella "Contract" per il deployment del contratto:
+\begin{itemize}
+  \item \textbf{mnemonic.secret}: deve contenere la mnemonic del proprio wallet;
+  \item \textbf{providerlink.secret}: deve contenere l'url del provider ottenibile come spiegato al punto \hyperref[sec:provider]{2.1.2}, può essere utilizzato anche l'url in http;
+  \item \textbf{apikey.secret}: deve contenere le api key ottenibili come spiegato al punto \hyperref[sec:provider]{2.1.2}.
```

Comment:
LA api key

Reply:
Poi in realtà no, la api key in quel file non è quella del provider. la api key in quel file la si ottiene andando su polygon scan e creando un account e copiando la chiave api, perchè serve solo a verificare il contratto su polygonscan. quella fornita dal provider sta in provider.secret nel link.